src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/LimitedUseItem.java

```diff
@@ -0,0 +1,141 @@
+ */
+public abstract class LimitedUseItem extends SimpleSlimefunItem<ItemUseHandler> {
+
+    private static final NamespacedKey usageKey = new NamespacedKey(SlimefunPlugin.instance(), "uses_left");
+
+    private int maxUseCount = -1;
+
+    @ParametersAreNonnullByDefault
+    public LimitedUseItem(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(category, item, recipeType, recipe, null);
+
+        addItemHandler(getItemHandler());
+    }
+
+    /**
+     * Returns the number of times this item can be used.
+     *
+     * @return The number of times this item can be used.
+     */
+    public final int getMaxUseCount() {
+        return maxUseCount;
+    }
+
+    /**
+     * Sets the maximum number of times this item can be used.
+     * The number must be greater than zero.
+     *
+     * @param count The maximum number of times this item can be used.
+     */
+    public final void setMaxUseCount(int count) {
+        Validate.isTrue(count > 0, "The maximum use count must be greater than zero!");
+
+        if (getState() == ItemState.UNREGISTERED) {
+            maxUseCount = count;
+        } else {
+            throw new IllegalStateException("You cannot modify the maximum use count after the Item was registered.");
+        }
+    }
+
+    /**
+     * Returns the {@link NamespacedKey} under which will the amount of uses left stored.
+     *
+     * @return The {@link NamespacedKey} to store/load the amount of uses
+     */
+    protected @Nonnull NamespacedKey getStorageKey() {
+        return usageKey;
+    }
+
+    @Override
+    public void register(@Nonnull SlimefunAddon addon) {
+        if (getMaxUseCount() < 1) {
+            warn("The use count has not been configured correctly. It needs to be at least 1. The Item was disabled.");
+        } else {
+            super.register(addon);
+        }
+    }
+
+    @ParametersAreNonnullByDefault
+    protected void damageItem(Player p, ItemStack item) {
+        if (item.getAmount() > 1) {
+            item.setAmount(item.getAmount() - 1);
+
+            // Separate one item from the stack and damage it
+            ItemStack separateItem = item.clone();
+            separateItem.setAmount(1);
+            damageItem(p, separateItem);
+
+            // Try to give the Player the new item
+            if (!p.getInventory().addItem(separateItem).isEmpty()) {
+                // or throw it on the ground
+                p.getWorld().dropItemNaturally(p.getLocation(), separateItem);
+            }
+        } else {
+            ItemMeta meta = item.getItemMeta();
+            NamespacedKey key = getStorageKey();
+            PersistentDataContainer pdc = meta.getPersistentDataContainer();
+            int usesLeft = pdc.getOrDefault(key, PersistentDataType.INTEGER, getMaxUseCount());
+
+            if (usesLeft == 1) {
+                p.playSound(p.getLocation(), Sound.ENTITY_ITEM_BREAK, 1, 1);
+                item.setAmount(0);
```

Comment:
Note: This doesn't *always* work as you expect. It can just set the amount to well 0. Which is technically a valid number in some versions of MC. Some server software on some versions have this behaviour rather than removing the item.

It's best to set the type to AIR.